File: solutions/julia/binary-search/1/binary-search.jl
# Change any of the following NamedTuple values to true to 
# enable testing different bonus tasks
tested_bonus_tasks = (rev = false, by = false, lt = false, multiple_matches = false)

# first iteration, no bonus tasks
function binarysearch(sorted_v, key)
    res = 0:0
    high = length(sorted_v)
    low = 1
    while low<=high
        mid = (high + low) ÷ 2
        if sorted_v[mid] > key
              high = mid-1
        elseif sorted_v[mid] < key
                low = mid+1
        elseif sorted_v[mid] == key
                res = mid:mid
                break 
            end
    end
    low>high && (res = low:high) 
    res   
end

File: solutions/python/binary-search/1/binary_search.py
def find(search_list, value):
    A = search_list
    high = len(A)-1
    low = 0
    while low<=high:
        mid = (high + low)//2
        if A[mid]<value:
            low = mid + 1
        elif A[mid]==value:
            return mid
        elif A[mid]>value:
            high = mid-1
    raise ValueError("value not in array")
        
    
